### PR TITLE
fix: episode thumbnail transition property

### PR DESCRIPTION
**Pull request by tokitou-san**

### CODE REVIEW

File: tailwind.config.ts [hunk 1]
```diff
@@ -4,7 +4,11 @@ const config: Config = {
     darkMode: "class",
     content: ["./src/**/*.{html,js,svelte,ts}", require("path").join(require.resolve("@skeletonlabs/skeleton"), "../**/*.{html,js,svelte,ts}")],
     theme: {
-        extend: {}
+        extend: {
+            transitionProperty: {
+                "max-height": "max-height"
+            }
+        }
```
Comment (baseplate-admin) [inline tailwind.config.ts:11]: Remove this

**Comment by baseplate-admin** [comment]
Consider using scroll_area for texts that are large :)

**Comment by tokitou-san** [comment]
**Current behavior with large content:**
[Screencast from 2023-06-17 21-24-45.webm](https://github.com/tokitou-san/CoreProject-V3-UI/assets/114811070/356d298c-e55c-4e20-b59e-3e450d8ea821)

**Comment by baseplate-admin** [comment]
> Current behavior with large content:

Looks 90% lgtm. Just make sure that japanese name and episode name each get their own `ScrollArea`

**Comment by tokitou-san** [comment]
Added `delay-1000` for preventing sudden shift in height when not hovering childern.
( Can be adjusted, but I think above 500 would be better )

**Comment by baseplate-admin** [comment]
Hola, Amigo, Como estas?

https://github.com/tokitou-san/CoreProject-V3-UI/assets/61817579/fe4399bd-bb0a-4d02-b41a-b08f39d292c2

**Comment by tokitou-san** [comment]
> Hola, Amigo, Como estas?
> Watch.Your.Lie.in.April.on.AnimeCore.Firefox.Nightly.2023-06-18.16-28-33.mp4

Ohh this motion remembers me something, uhh whats that :|

**Comment by baseplate-admin** [comment]
> Ohh this motion remembers me something

you need help

**Comment by tokitou-san** [comment]
@baseplate-admin 
Still need to refactor, styling is causing main issues
Pushed in case if you want to take a look

**Main update**:
Max height of the container updates dynamically according to the height of hover element.
But `japanese-name` looks off

[Screencast from 2023-06-20 01-21-12.webm](https://github.com/tokitou-san/CoreProject-V3-UI/assets/114811070/ac9bfac7-73b2-40e6-a68e-a5fde03e4cb3)

File: src/lib/components/pages/anime_info/index.svelte [hunk 2]
```diff
@@ -761,13 +822,10 @@
...
-            /* if we need to change the width, we should change the 90% to higher  */
-            mask-image: linear-gradient(90deg, rgba(7, 5, 25, 0.95) 90%, rgba(0, 0, 0, 0) 100%);
-            mask-repeat: no-repeat;
-            mask-position: right;
+    // tailwind trasnitions seems not working
+    @media (min-width: 768px) {
+        episode-info-card {
+            transition: max-height 0.2s ease-in;
         }
     }
```
Comment (baseplate-admin) [inline src/lib/components/pages/anime_info/index.svelte:830]: Remove this

File: src/lib/components/pages/anime_info/index.svelte [hunk 3]
```diff
@@ -373,18 +406,46 @@
...
+                                                    </episode-name>
+                                                </div>
+                                            </scroll-area>
+                                            <scroll-area
+                                                id="scroll-area-{index}"
+                                                class="pointer-events-auto h-full max-h-9 w-full md:absolute md:bottom-[3.5vw] md:max-h-[1vw] md:overflow-hidden md:hover:max-h-full"
+                                                on:mouseenter|stopPropagation|self={() => {
+                                                    handle_mouseenter();
+                                                    hovered_type = "japanese_title";
+                                                }}
```
Comment (baseplate-admin) [inline src/lib/components/pages/anime_info/index.svelte:435]: Consider changing `hovered_type` inside `mouseenter`

File: src/lib/components/pages/anime_info/index.svelte [hunk 4]
```diff
@@ -373,18 +406,46 @@
...
+                                        class="pointer-events-none relative col-span-7 flex h-full w-full flex-col items-start justify-between md:absolute md:bottom-0 md:col-span-12 md:rounded-b-[0.625vw] md:bg-surface-900 md:p-[1vw]"
+                                        style="max-height: {episode_info_heights[index]}vw;"
+                                    >
+                                        <div class="relative flex h-full w-full flex-col items-start gap-1 md:gap-[0.5vw]">
+                                            <scroll-area
+                                                id="scroll-area-{index}"
+                                                class="pointer-events-auto h-full max-h-9 md:max-h-[1vw] md:overflow-hidden md:hover:max-h-full"
+                                                on:mouseenter|stopPropagation|self={() => {
+                                                    handle_mouseenter();
+                                                    hovered_type = "title";
```
Comment (baseplate-admin) [inline src/lib/components/pages/anime_info/index.svelte:419]: Consider changing hovered_type inside mouseenter

File: src/lib/components/pages/anime_info/index.svelte [hunk 5]
```diff
@@ -344,22 +370,29 @@
...
                                 {@const episode_number = episode.number}
                                 {@const japanese_name = episode.japanese_title}
                                 {@const duration = episode.duration}
 
+                                {@const handle_mouseenter = () => {
+                                    handle_episode_hover(index);
+                                }}
+                                {@const handle_mouseleave = () => {
+                                    episode_info_heights[index] = 8;
+                                }}
```
Comment (baseplate-admin) [inline src/lib/components/pages/anime_info/index.svelte:385]: Unnecessary

File: src/lib/components/pages/anime_info/index.svelte [hunk 6]
```diff
@@ -105,6 +105,32 @@
...
+            }
+        } else if (hovered_type === "japanese_title") {
+            let element = document.querySelector(`#japanese-name-${index}`) as HTMLElement;
+
+            if (element) {
+                const height_in_vw = (element.offsetHeight / window.innerWidth) * 100;
+                console.log(episode_info_heights[index] + Math.floor(height_in_vw));
+                episode_info_heights[index] = episode_info_heights[index] + Math.floor(height_in_vw);
+            }
+        }
```
Comment (baseplate-admin) [inline src/lib/components/pages/anime_info/index.svelte:132]: Switch case is better for this

**Review by baseplate-admin — CHANGES_REQUESTED** [review]
Need some logic refractor

File: src/lib/components/pages/anime_info/index.svelte [hunk 7]
```diff
@@ -761,13 +791,21 @@
...
+                overflow-y: hidden;
+                height: auto;
+                max-height: 1vw;
+
+                &:hover {
+                    max-height: var(--max-height-hover, 3.75vw);
+                }
+            }
         }
     }
```
Comment (baseplate-admin) [inline src/lib/components/pages/anime_info/index.svelte:810]: We should convert it to tailwind classes :)

want me to work on this?

Comment (tokitou-san) [reply src/lib/components/pages/anime_info/index.svelte]: >We should convert it to tailwind classes :)
want me to work on this?

Yea duh, but you want to refactor js?

**Comment by baseplate-admin** [comment]
This is honestly one of the hardest thing we have achieved. Well done

**Pull request merged**